### DISCUSSION TITLE
How can two models contain Model types of each other? (chicken egg problem)

### DISCUSSION THREAD

**Discussion by Prinzhorn** [dynamoose/dynamoose#1095] [Q&A] [answered]
I have two models, say `User` and `Presentation`. I want a `presentations` list on the `User` and an `owner` on the `Presentation`. How do I do that when each schema needs to reference the other model, but the model needs the schema to be created?

In Mongoose I can do this (copied from an old project, maybe it has changed):

```
owner: {
  type: Schema.Types.ObjectId,
  ref: 'User',
}
```

If I remember correctly the existence of `User` wasn't validated at `mongoose.model('Presentation', presentationSchema)` but when the model was first used. This would allow setting up all the references without worrying about the order in which you define the schemas and models.

I'm probably just missing something on https://dynamoosejs.com/guide/Schema#model-types

**Comment by fishcharlie** [comment, marked as answer]
@Prinzhorn I don't think this is currently possible in Dynamoose. What is your use case here? Is there a reason you need a bidirectional relationship in this way? In terms of retrieving data, what use case does the enable?

Normally when discussing DynamoDB a lot of SQL design patterns just aren't efficient in DynamoDB. The `populate` function in Dynamoose isn't really meant to be efficient, but to provide a stopgap to make it easy to convert to. Ideally to take advantage of the performance benefits of DynamoDB you wouldn't use it.

I'd suggest you watch https://youtu.be/DIQVJqiSUkE. It has a lot of great information about modeling your data in DynamoDB. All of the features discussed should be possible in Dynamoose.

Now if you really want to have a workaround for this, you can just store the key as an attribute in your table, then use a `get` or `query` call to retrieve the other item. Behind the scenes that is what Dynamoose is doing for this functionality. So even tho this isn't possible in Dynamoose currently, you can easily add your own workaround for it using your own system.

If you feel like this feature should be added feel free to create an issue on this repo with a feature request for it.

Hopefully this helps answer your question. Sorry this isn't possible currently.

**Reply by Prinzhorn** [reply]
Thanks for your answer. I'm working on one of those projects were budget dictates everything and I don't care about if the design is "good". I'm not using DynamoDB for any of its properties, I'm using it because it's integrated with Amplify and doesn't require much setup. At one point I even considered putting _all_ of the application data in a single document and doing everything in the application. The main reason I didn't was that the size _might_ outgrow 400KB at some point (probably not) and I want to avoid more pain in the future :sweat_smile: 

> What is your use case here? Is there a reason you need a bidirectional relationship in this way? In terms of retrieving data, what use case does the enable?

I guess my thought was that this is a nicer way to access the data than using secondary indexes manually. But I guess I could make that look nicer and implicitly "populate" using a `get` on a virtual `Users.presentations` that queries the `ownerId` index.
